api/candlepin-api-spec.yaml

```diff
@@ -555,239 +555,1732 @@ paths:
+        404:
+          description: Consumer with this UUID could not be found.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ExceptionMessage'
+              example:
+                displayMessage: Consumer with this UUID could not be found.
+                requestUuid: c4347004-8792-41fe-a4d8-fccaa0d3898a
+        default:
+          $ref: '#/components/responses/default'
+
+  /consumers/{consumer_uuid}/certificates:
+    put:
+      description: Regenerates the Entitlement Certificates for a Consumer
+      tags:
+        - Consumer
+        - Certificates
+      operationId: regenerateEntitlementCertificates
+      security: [ ]
+      parameters:
+        - name: consumer_uuid
+          in: path
+          description: The UUID of the Consumer
+          required: true
+          schema:
+            type: string
+        - name: entitlement
+          in: query
+          description: Entitlement ID
+          schema:
+            type: string
+        - name: lazy_regen
+          in: query
+          description: Lazy regeneration of entitlement certs
+          schema:
+            type: boolean
+            default: true
+      responses:
+        200:
+          description: Successful operation
+        400:
+          description: Consumer does not have a defined type ID. Invalid consumer type.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ExceptionMessage'
+              example:
+                displayMessage: Consumer does not have a defined type ID. Invalid consumer type.
+                requestUuid: c4347004-8792-41fe-a4d8-fccaa0d3898a
+        404:
+          description: Consumer with this UUID could not be found Or Entitlement with this ID could not be
+            found.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ExceptionMessage'
+              example:
+                displayMessage: Consumer with this UUID could not be found Or Entitlement with this
+                  ID could not befound.
+                requestUuid: c4347004-8792-41fe-a4d8-fccaa0d3898a
+        default:
+          $ref: '#/components/responses/default'
+
+    get:
+      description: Retrieves a Compressed File of Entitlement Certificates
+      tags:
+        - Consumer
+        - Certificates
+      operationId: exportCertificates
+      security: [ ]
+      parameters:
+        - name: consumer_uuid
+          in: path
+          description: The UUID of the consumer to retrieve guest
+          required: true
+          schema:
+            type: string
+        - name: serials
+          in: query
+          description: Certificate serials
+          schema:
+            type: string
+      responses:
+        200:
+          description: A successful operation
+          content:
+            text/plain:
```

Comment:
The response type here directly translate into method return type. So ``application/zip`` would be ``File`` &  ``application/json`` would be ``CertificateDTO``. Since we merged the APIs, I added ``text/plain`` which translate to ``Object`` return type. I couldn't find anything better.

Reply:
Yes it looks like openapi-generator does not support this behaviour. Let's leave this as is then, but add a comment right above this line that explains that this is a workaround due to openapi-generator not supporting multiple schemas for different accept header values on the same endpoint

Reply:
Done.

Reply:
I think we should be even more detailed here, for the sake of people reading this in the future. `This endpoint does not really support text/plain, but only application/zip and application/json. The reason for adding text/plain here is a workaround and it is because only the first schema type defined is taken into account. In this case Object will be generated as the returned type, which covers both the List<CertificateDTO> and zip file of the 2 cases we support. This is done because currently openapi-generator does not support returning multiple schema types for different accept header values on the same endpoint`

Reply:
Done.